src/applications/vaos/tests/new-appointment/components/VAFacilityPage/VAFacilityPageV2.multi.unit.spec.jsx

```diff
@@ -253,6 +254,95 @@ describe('VAOS integration: VA flat facility page - multiple facilities', () =>
     expect(firstRadio).to.contain.text('Closest facility');
   });
 
+  it('should sort by distance from current location if user clicks "use current location"', async () => {
+    mockParentSites(parentSiteIds, [parentSite983, parentSite984]);
+    mockDirectBookingEligibilityCriteria(parentSiteIds, directFacilities);
+    mockRequestEligibilityCriteria(parentSiteIds, requestFacilities);
+    mockFacilitiesFetch(vhaIds.join(','), facilities);
+    mockEligibilityFetches({
+      siteId: '983',
+      facilityId: '983',
+      typeOfCareId: '323',
+    });
+    mockGetCurrentPosition();
+    const store = createTestStore({
+      ...initialState,
+      user: {
+        ...initialState.user,
+        profile: {
+          ...initialState.user.profile,
+          vet360: {
+            residentialAddress: {
+              addressLine1: '290 Ludlow Ave',
+              city: 'Cincinatti',
+              stateCode: 'OH',
+              zipCode: '45220',
+              latitude: 39.1362562, // Cincinatti, OH
+              longitude: -84.6804804,
+            },
+          },
+        },
+      },
+    });
+    await setTypeOfCare(store, /primary care/i);
+
+    const screen = renderWithStoreAndRouter(<VAFacilityPage />, {
+      store,
+    });
+
+    await screen.findAllByRole('radio');
+    fireEvent.click(screen.getByText('use your current location'));
+    await screen.findByText('Finding your location...');
```

Comment:
I would skip checking for the loading message in these tests. Based on the flaky tests in Jenkins, it doesn't seem like we can always rely on catching this text before it disappears.

This seems like a limitation of the fetch mocking approach we use, where we can't hold open a request and resolve it when we choose. I might put in a ticket to see if there's a way to build that functionality so we can reliably test loading states.

Reply:
Thanks for the heads up. Updated to look for radio buttons to reappear instead before checking for updated text.